### PR TITLE
Split Hessian functions into two removing multipliers as keywords

### LINKED ISSUE #211: Remove Lagrangian multipliers as kw from Hessian functions splitting into two functions
Instead of `hess(nlp, x, y=y)` have `hess(nlp, x)` and `hess(nlp, x)` and `hess(nlp, x, y)` are different functions.

### PR DESCRIPTION
**Pull request by abelsiqueira**
~Blocked by #212, don't review yet.~

Closes #211

### CODE REVIEW

**Comment by abelsiqueira** [comment]
:robot: Testing breakage of this commit

| Package Name | master | Tagged |
|--|--|--|
| AmplNLReader |  |  |
| CaNNOLeS |  |  |
| CUTEst |  |  |
| QuadraticModels |  |  |
| NLPModelsIpopt |  |  |
| NLPModelsJuMP |  |  |
| SolverTools |  |  |

**Comment by abelsiqueira** [comment]
:robot: Testing breakage of this commit

| Package Name | master | Tagged |
|--|--|--|
| AmplNLReader |  |  |
| CaNNOLeS |  |  |
| CUTEst |  |  |
| QuadraticModels |  |  |
| NLPModelsIpopt |  |  |
| NLPModelsJuMP |  |  |
| SolverTools |  |  |

**Comment by abelsiqueira** [comment]
:robot: Testing breakage of this commit

| Package Name | master | Tagged |
|--|--|--|
| AmplNLReader |  |  |
| CaNNOLeS |  |  |
| CUTEst |  |  |
| QuadraticModels |  |  |
| NLPModelsIpopt |  |  |
| NLPModelsJuMP |  |  |
| SolverTools |  |  |

**Comment by abelsiqueira** [comment]
:robot: Testing breakage of this commit

| Package Name | master | Tagged |
|--|--|--|
| AmplNLReader |  |  |
| CaNNOLeS |  |  |
| CUTEst |  |  |
| QuadraticModels |  |  |
| NLPModelsIpopt |  |  |
| NLPModelsJuMP |  |  |
| SolverTools |  |  |

File: src/NLPModels.jl [hunk 1]
```diff
@@ -352,82 +358,177 @@ Return the structure of the Lagrangian Hessian in sparse coordinate format in pl
...
+
+Return the objective Hessian at `x` with objective function scaled by
+`obj_weight` as a linear operator. The resulting object may be used as if it were a
+matrix, e.g., `H * v`. The linear operator H represents
+$(OBJECTIVE_HESSIAN).
+"""
+function hess_op(nlp :: AbstractNLPModel, x :: AbstractVector; obj_weight::Real=one(eltype(x)))
+  prod = @closure v -> hprod(nlp, x, v; obj_weight=obj_weight)
+  F = typeof(prod)
+  return LinearOperator{Float64,F,F,F}(nlp.meta.nvar, nlp.meta.nvar,
```
[the remark this answers is not in the document]
Comment (abelsiqueira) [reply src/NLPModels.jl]: I can limit the LinearOperators version to 0.6 (I imagine the next LinearOperators release with #99 would be 0.7 -- or 1.0).
Or the opposite, and require the next LinearOperators.

[the remark this answers is not in the document]
Comment (dpo) [reply src/NLPModels.jl]: Yes, let's limit the LinearOperators version. The next version of NLPModels will require LinearOperators 0.7 or 1.0.

**Comment by abelsiqueira** [comment]
:robot: Testing breakage of this commit

| Package Name | master | Tagged |
|--|--|--|
| AmplNLReader |  |  |
| CaNNOLeS |  |  |
| CUTEst |  |  |
| QuadraticModels |  |  |
| NLPModelsIpopt |  |  |
| NLPModelsJuMP |  |  |
| SolverTools |  |  |

File: src/NLPModels.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (abelsiqueira) [reply src/NLPModels.jl]: Done

**Comment by dpo** [comment]
Great, thanks. Feel free to merge when the tests pass.

**Pull request merged**